block/internal/syncing/syncer.go

```diff
@@ -349,6 +351,8 @@ func (s *Syncer) tryFetchFromP2P() {
 
 	// Process data (if not already processed by headers)
 	newDataHeight := s.dataStore.Store().Height()
+	// TODO @MARKO: why only if newDataHeight != newHeaderHeight? why not process
```

Comment:
in https://github.com/evstack/ev-node/pull/2798 we refactor this code making it simpler and more robust. We can merge this pr in order to solve the hashing issue